app.js

```diff
@@ -1,41 +1,53 @@
 import { app, errorHandler } from 'mu';
 import { SERVICE_NAME } from './lib/env';
-import { getScheduledDumpTask, produceDumpFile } from './lib/helpers';
-import { getLatestFile } from './lib/queries';
+import { getScheduledDumpTask } from './lib/helpers';
+import { storeError } from './lib/queries';
+import { updateStatus, appendTaskError } from './lib/queries';
+import { STATUS_BUSY, STATUS_SUCCESS, STATUS_FAILED } from './lib/constants';
+import { DatasetManager } from './lib/dataset';
 import bodyParser from 'body-parser';
 import flatten from 'lodash.flatten';
-import path from 'path';
-import fs from 'fs-extra';
 
-app.use(bodyParser.json({ type: function(req) { return /^application\/json/.test( req.get('content-type') ); } }))
+app.use(bodyParser.json({ type: function(req) { return /^application\/json/.test( req.get('content-type') ); } }));
 
-app.get('/', function(req, res) {
+app.get('/', function(_, res) {
   const message = `Hey there, you have reached ${SERVICE_NAME}! Seems like I'm doing just fine :)`;
   res.send(message);
 });
 
 app.post('/delta', async function( req, res ) {
-  const delta = req.body;
-  const inserts = flatten(delta.map(changeSet => changeSet.inserts));
-  const task = await getScheduledDumpTask(inserts);
-  if (task) {
-    produceDumpFile(task); // Not awaiting to avoid socket hangup in deltanotifier
-    res.send({message: `Dump file production started`});
-  } else {
-    console.log('Incoming deltas do not contain any busy job, skipping.');
-    res.status(204).send();
+  try {
+    const delta = req.body;
+    const inserts = flatten(delta.map(changeSet => changeSet.inserts));
+    const task = await getScheduledDumpTask(inserts);
+    if (task) {
+      produceDumpFile(task); // Not awaiting to avoid socket hangup in deltanotifier
+      res.send({message: `Dump file production started`});
+    }
+    else {
+      console.log('Incoming deltas do not contain any busy job, skipping.');
+      res.status(204).send();
+    }
+  }
+  catch(e){
+    const msg = `General error with creating dump file: ${e}`;
+    console.error(msg);
+    await storeError(msg);
   }
 });
 
-app.get('/latest-dump-file', async (req, res) => {
```

Comment:
Why are we deleting this ? It changes a bit the analysis of mandatendatabank-consumer then

Reply:
so here, mu-resource + file service will do the serving. With one API call you should get the information what you need. 
`/datasets?filter[:has-no:next-version]=yes`